sdk/eventhub/azure-eventhub/azure/eventhub/_configuration.py

```diff
@@ -51,3 +56,41 @@ def __init__(self, **kwargs):
             self.connection_port = endpoint.port or DEFAULT_AMQP_WSS_PORT
+
+
+class PartitionPublishingConfiguration(DictMixin):
+    """
+    The set of configurations that can be specified for an `EventHubProducerClient`
+    to influence its behavior when publishing directly to an Event Hub partition.
+
+    :ivar int producer_group_id: The identifier of the producer group that this producer is associated with when
+     publishing to the associated partition.
+    :ivar int owner_level: The owner level indicates that a publishing is intended to be performed exclusively for
+     events in the requested partition in the context of the associated producer group.
+    :ivar int starting_sequence_number: The starting number that should be used for the automatic sequencing of
+     events for the associated partition, when published by this producer.
+
+    :keyword int producer_group_id: The identifier of the producer group that this producer is associated with when
+     publishing to the associated partition. The producer group is only recognized and relevant when certain features
+     of the producer are enabled. For example, it is used by idempotent publishing.
+     The producer group id should be in the range from 0 to max signed long (2^63 - 1) as required by the service.
+    :keyword int owner_level: The owner level indicates that a publishing is intended to be performed exclusively for
+     events in the requested partition in the context of the associated producer group. To do so, publishing will
+     attempt to assert ownership over the partition; in the case where more than one publisher in the producer
+     group attempts to assert ownership for the same partition, the one having a larger owner_level value will "win".
+     The owner level is only recognized and relevant when certain features of the producer are enabled. For example,
+     it is used by idempotent publishing.
+     The owner level should be in the range from 0 to max signed short (2^16 - 1) as required by the service.
+    :keyword int starting_sequence_number: The starting number that should be used for the automatic sequencing of
+     events for the associated partition, when published by this producer. The starting sequence number is only
+     recognized and relevant when certain features of the producer are enabled. For example, it is used by idempotent
+     publishing.
+     The starting sequence number should be in the range from 0 to max signed integer (2^31 - 1) as
+     required by the service.
+
+    """
+    def __init__(self, **kwargs):
+        validate_producer_client_partition_config(kwargs)
+        self.owner_level = kwargs.get("owner_level")  # type: Optional[int]
+        self.producer_group_id = kwargs.get("producer_group_id")  # type: Optional[int]
+        self.starting_sequence_number = kwargs.get("starting_sequence_number")  # type: Optional[int]
```

Comment:
I thought all these were required?

Reply:
AKAIK those are optional for the advanced creation scenario, have played with the different configs.
If those settings are mis-configured, the service would return kinda "invalid state" error.